### ISSUE TITLE
SLIP-0013 hash validity

### ISSUE THREAD

**Issue by joaojoyce** [satoshilabs/slips#107]
Hi,

I'm trying to implement https://github.com/satoshilabs/slips/blob/master/slip-0013.md in PHP but I'm having some problems with the example presented:

In particular I'm not finding the correct result for this example:
````
index + uri=0 + https://satoshi@bitcoin.org/login
sha256(index + uri) = d0e2389d4c8394a9f3e32de01104bf6e8db2d9e2bb0905d60fffa5a18fd696db
````

In my PHP code I'm doing:
````
    $url = "https://satoshi@bitcoin.org/login";
    $index = 0;
    
    //This gives me the hash 59550cf1de0625bcc6a368e5fd3395c917d6e368fa8b908df96797f349d10e02

    echo hash("sha256",$index . $url);

````

Could someone confirm the correctness of the example? Could this be an encoding problem or something similar?

Thank you very much.

**Comment by prusnak** [comment]
You need to use 32-bit unsigned integer as index. Your code treats that as string.

```
pack("L" $index) . $url 
```

should work

**Comment by joaojoyce** [comment]
Thank you very much. That worked perfectly. I managed to finish the example but I still found something strange. 

Although I found the A' B' C' D' correctly the integers for A, B, C, D in the example don't match mine.

Example:
`````
//These don't match the ones I got
A=221831888
B=160727884
C=3007475
D=247399441

//These match perfectly
A'=2637750992
B'=2845082444
C'=3761103859
D'=4005495825

`````

Mine:
`````
A=2637750992
B=2845082444
C=3761103859
D=1858012177

A'=2637750992
B'=2845082444
C'=3761103859
D'=4005495825
`````

**Comment by java4tech** [comment]
Hi,
I'm also trying to implement it in java. I'm also facing same issue with the hash. I looked at PHP example and following solution.

> You need to use 32-bit unsigned integer as index. Your code treats that as string.
> The following should work:
> pack("V", $index) . $url

But I could not get the right hash in Java.  any help will be nice.

Thanks

**Comment by andrewkozlik** [comment]
Fixed in 4131f57
